File: rest-common/src/main/java/org/chronopolis/rest/models/repair/RepairStatus.java
package org.chronopolis.rest.models.repair;

import com.google.common.collect.ImmutableListMultimap;

/**
 * Enumeration of the various statuses our repairs can have
 *
 * Created by shake on 11/10/16.
 */
public enum RepairStatus {
    REQUESTED, STAGING, READY, TRANSFERRED, REPAIRED, FAILED;

    public static ImmutableListMultimap<String, RepairStatus> statusByGroup() {
        return new ImmutableListMultimap.Builder<String, RepairStatus>()
                .put("Pending", RepairStatus.REQUESTED)
                .put("Active", RepairStatus.STAGING)
                .put("Active", RepairStatus.READY)
                .put("Active", RepairStatus.TRANSFERRED)
                .put("Success", RepairStatus.REPAIRED)
                .put("Failure", RepairStatus.FAILED)
                .build();
    }
}

File: rest-models/src/main/kotlin/org/chronopolis/rest/models/enums/RepairStatus.kt
package org.chronopolis.rest.models.enums

import com.google.common.collect.ImmutableListMultimap

enum class RepairStatus {
    REQUESTED, STAGING, READY, TRANSFERRED, REPAIRED, FAILED;

    companion object {
        fun statusByGroup(): ImmutableListMultimap<String, RepairStatus> =
                ImmutableListMultimap.Builder<String, RepairStatus>()
                        .put("Pending", REQUESTED)
                        .put("Active", STAGING)
                        .put("Active", READY)
                        .put("Active", TRANSFERRED)
                        .put("Success", REPAIRED)
                        .put("Failure", FAILED)
                        .build()
    }
}